Lib/http/client.py

```diff
@@ -941,23 +941,26 @@ def _tunnel(self):
         del headers
 
         response = self.response_class(self.sock, method=self._method)
-        (version, code, message) = response._read_status()
+        try:
+            (version, code, message) = response._read_status()
 
-        if code != http.HTTPStatus.OK:
-            self.close()
-            raise OSError(f"Tunnel connection failed: {code} {message.strip()}")
-        while True:
-            line = response.fp.readline(_MAXLINE + 1)
-            if len(line) > _MAXLINE:
-                raise LineTooLong("header line")
-            if not line:
-                # for sites which EOF without sending a trailer
-                break
-            if line in (b'\r\n', b'\n', b''):
-                break
+            if code != http.HTTPStatus.OK:
+                self.close()
```

Comment:
There is `self.close()` raised ahead of OSError.
So, it should have closed the connection. Not sure if `response.close()` was expected here.  https://github.com/python/cpython/blob/main/Lib/http/client.py#L947

Reply:
Yes response.close() is expected here, Response takes a reference to the socket with

```
self.fp = sock.makefile()
```

And that means even though `sock.close()` is called it's held open by the makefile